lib/builders/BrowserBundleBuilder.js

```diff
@@ -289,7 +317,79 @@ class BrowserBundleBuilder {
 			});
 
 		resetHandler(); // to set startTime universally.
-		return this._bundler;
+		return this._appBundler;
+	}
+
+	/**
+	 * Creates the browserify bundler for externals or re-uses the existing one.
+	 * @returns {Browserify} The browserify instances.
+	 * @private
+	 */
+	_createExternalsBundler() {
+		if (this._externalsBundler) {
+			return this._externalsBundler;
+		}
+
+		// const filenames = Object.keys(this._externalModules)
+		//				.map(id => require.resolve(path.join(WORKING_DIR, 'node_modules', id)));
+
+		this._externalsBundler = browserify({
+			cache: {},
+			packageCache: {},
+			debug: !this._isRelease
+		});
+		this._externalsBundler.require(Object.keys(this._externalModules));
+		this._externalsBundler.external('appDefinitions');
+		this._attachExtensionsToBundler(this._externalsBundler);
+
+		const startTime = hrTimeHelper.get();
+		this._eventBus.emit('info', `Building browser script bundle for externals at "${this._externalsPath}"...`);
+
+		this._externalsBundler
+			.on('error', error => this._eventBus.emit('error', error))
+			.on('bundle', sourceStream => {
+				const outputStream = fs.createWriteStream(this._externalsPath);
+				outputStream.once('finish', () => {
+					const hrTime = hrTimeHelper.get(startTime);
+					this._eventBus.emit('externalsBundleBuilt', {
+						path: this._externalsPath,
+						hrTime,
+						time: hrTimeHelper.toMilliseconds(hrTime)
+					});
+				});
+				sourceStream.pipe(outputStream);
+			});
+
+		return this._externalsBundler;
+	}
+
+	/**
+	 * Attaches necessary plugins and transformations to the bundle.
+	 * @param {Browserify} bundler The bundler to attach extensions.
+	 */
+	_attachExtensionsToBundler(bundler) {
+		bundler.transform(babelify, {
```

Comment:
Can't remove completely even for the debug mode, people use a lot of new JS features browsers still don't support. I will integrate https://github.com/babel/babel-preset-env before the next release and it would have a configuration option.